src/components/logs/Context.tsx

```diff
@@ -0,0 +1,174 @@
+import {
+    createContext,
+    useContext,
+    useEffect,
+    useCallback,
+    useState,
+    useRef,
+} from 'react';
+import { BaseComponentProps, ViewLogs_Line } from 'types';
+import { DEFAULT_POLLING_INTERVAL, RPCS } from 'services/supabase';
+import { hasLength, incrementInterval, timeoutCleanUp } from 'utils/misc-utils';
+import { useClient } from 'hooks/supabase-swr';
+
+interface Props extends BaseComponentProps {
+    token: string | null;
+    disableIntervalFetching?: boolean;
+    fetchAll?: boolean;
+}
+
+type LogsContextValue = {
+    reset: () => void;
+    logs: ViewLogs_Line[];
+    networkFailure: boolean;
+    stopped: boolean;
+};
+
+const MAX_EMPTY_CALLS = 90; // about 1.5 minute
+const MAX_INTERVAL = 1000; //  we don't want to wait too longer between pings
+const LogsContext = createContext<LogsContextValue | null>(null);
+
+const LogsContextProvider = ({
```

Comment:
No they shouldn't. If context (or stores or hooks) are really only for one section it is fine keeping them close to what consumes them. Since this is a context that is nested deep into the component it is fine keeping it close to it.